xbmc/peripherals/bus/PeripheralBus.h

```diff
@@ -54,7 +54,7 @@ namespace PERIPHERALS
     /*!
      * @return True if this bus needs to be polled for changes, false if this bus performs updates via callbacks
      */
-    bool NeedsPolling(void) const { return m_bNeedsPolling; }
+    bool NeedsPolling(void) const { CSingleLock lock(m_critSection); return m_bNeedsPolling; }
```

Comment:
Whichever you think, both works. You don't need to call compare_exchange_strong manually, it's done for you in the assignment and conversion operators by specifying the memory barrier type during declaration. So what you get is a bool that you treat as any other bool but it's thread safe under the covers, Which is the beauty of std::atomic. For int ++ and - - are overloaded as well so they're thread safe